x-pack/plugin/security/qa/jwt-realm/build.gradle

```diff
@@ -10,7 +10,13 @@ dependencies {
   javaRestTestImplementation project(":client:rest")
 }
 
-boolean explicitIdTokenType = BuildParams.random.nextBoolean()
+def random = BuildParams.random
+boolean explicitIdTokenType = random.nextBoolean()
+def serviceSubject = 'service_' + random.nextInt(1, 9) + '@app' + random.nextInt(1, 9) + '.example.com'
+
+tasks.named("javaRestTest").configure {
+  systemProperty 'jwt2.service_subject', serviceSubject
+}
```

Comment:
This is to make the randomly generated service principal available in the JavaRestTest. Please let me know if there is a better way to do this.